### PR TITLE
introduce infra e2e

### PR DESCRIPTION
**Pull request by dmsolr**

### CODE REVIEW

**Review by wu-sheng — COMMENTED** [review]
We have e2e tests hosted on OAP GitHub docker repo. Could you consider using them, and remove Java source codes from this repo? Is this possible?

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -30,15 +30,33 @@ jobs:
     timeout-minutes: 180
     steps:
       - uses: actions/checkout@v2
```
Comment (kezhenxu94) [inline .github/workflows/e2e.yaml:32]: Let's upgrade to v3 too

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -30,15 +30,33 @@ jobs:
...
-      - name: checkout submodules
-        shell: bash
-        run: |
-          git submodule sync --recursive
-          git -c protocol.version=2 submodule update --init --force --recursive --depth=1
+        with:
+          submodules: true
       - uses: actions/setup-java@v1
```
Comment (kezhenxu94) [inline .github/workflows/e2e.yaml:38]: Let's upgrade to v3 and use the maven cache

**Review by kezhenxu94 — APPROVED** [review]

**Comment by dmsolr** [comment]
> We have e2e tests hosted on OAP GitHub docker repo. Could you consider using them, and remove Java source codes from this repo? Is this possible?

I am confused as well. Because the old test framework has a limitation when i want to add a test case for request-id as trace id.
the infra e2e framework seem like it can cover the old test to verify trace info.

**Comment by wu-sheng** [comment]
We began to host test service after you set this repo.

**Comment by dmsolr** [comment]
introduce infra-e2e framework seems over-testing. therefore, I raised a new PR to improve the old testing.

**Pull request closed without merging**